src/main/java/com/google/cloud/tools/libraries/CloudLibraries.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.libraries;
+
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/** Returns helpful metadata for supported Google Cloud libraries. */
+public final class CloudLibraries {
+
+  private static final String LIBRARIES_JSON = "libraries.json";
+
+  private final String librariesJsonPath;
+
+  @VisibleForTesting
+  CloudLibraries(String librariesJsonPath) {
+    this.librariesJsonPath = librariesJsonPath;
+  }
+
+  /**
+   * Returns the list of {@link CloudLibrary} objects deserialized from the {@code libraries.json}
+   * file.
+   *
+   * @throws IOException if there was a problem reading the {@code libraries.json} file
+   */
+  public static List<CloudLibrary> getCloudLibraries() throws IOException {
+    return new CloudLibraries(LIBRARIES_JSON).getLibraries();
+  }
+
+  @VisibleForTesting
+  List<CloudLibrary> getLibraries() throws IOException {
+    try (InputStream inputStream = CloudLibraries.class.getResourceAsStream(librariesJsonPath)) {
+      if (inputStream == null) {
+        throw new IOException("Resource not found: " + librariesJsonPath);
+      }
+
+      InputStreamReader reader = new InputStreamReader(inputStream);
+      JsonReader jsonReader = new JsonReader(reader);
+      Type listType = new TypeToken<List<CloudLibrary>>() {}.getType();
+      return new Gson().fromJson(jsonReader, listType);
```

Comment:
This tightly couples the JSON data to the Java class, while still defining each independently. That's where brittleness comes from.

Reply:
Maybe I'm missing something, but the point of this class is to represent the data in the JSON file in a structured way. The file and this class are designed to be coupled together, but I disagree that the coupling is so tight that it would cause *brittleness* with changes made to either. If you change the JSON (e.g. new keys, remove old keys, etc.), the model class can still be built without error. What alternative would you suggest?

Reply:
Think of it this way: ask for what you need, not for what the supplier has. CloudLibraries in CT4E takes a very different approach.

Reply:
@elharo I'm also a little confused by what you are suggesting. Are you suggesting eclipse and intellij duplicate this logic? I assume CT4E has logic to parse this json file, just like CT4IJ will need.

Reply:
> If you change the JSON (e.g. new keys, remove old keys, etc.), the model class can still be built without error.

Nit-picking: it's not always the case. For example, if you change an int property to a string, or a string property to an array, `fromJson()` will throw a (baffling) malformed-JSON runtime exception. However, I admit this kind of changes would be rare. Also, if the point of this class is to be tightly coupled together by its design, I guess talking about brittleness is irrelevant anyway IMO.

FYI, the approach in Eclipse is (conceptually and supposed to be) like querying using JsonPath (an XPath counterpart for JSON) or such (although in reality, this is implemented in a naive way).

Reply:
I don't object to @nkibler's approach here.  We're already happily using GSON databinding with the CloudSdkComponent and in IntelliJ.   There can be issues using object mapping at scale on large class hierarchies.  I don't see a risk of that happening here.

Reply:
@chanseokoh that's correct, it will break if the types are incompatible. I added a unit test to explicitly call this out for clarity.